bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java

```diff
@@ -147,6 +159,26 @@ public Set<String> configure()
             logging = Logging.initialize();
         }
 
+        if (loadConfigResolver) {
+            log.info("Loading configuration resolver");
+            Map<String, String> configurationResolverProperties = Collections.emptyMap();
+            String configResolverFile = System.getProperty("configResolver");
+            if (configResolverFile != null) {
```

Comment:
`configurationResolver` will be null if the config property is not set, which seems wrong

Reply:
We'll want to allow configuring multiple providers, as people may have secrets in multiple systems. While this is rare in the common case, it is likely to be useful or necessary during a migration from one system to another.

What do you think about making this file be TOML similar to #1189? Then we could support multiple providers in a single file.

Reply:
Another issue... do we need to support environment variable interpolation in the `config_resolver` file? I have a feeling that this will be needed, unfortunately.

Reply:
> We'll want to allow configuring multiple providers, as people may have secrets in multiple systems. While this is rare in the common case, it is likely to be useful or necessary during a migration from one system to another.

We currently handle multiple systems - In the configResolver file we track the list of config providers we need to use. But yes using TOML would make it easy to configure.

Reply:
> configurationResolver will be null if the config property is not set, which seems wrong. 

We would be replacing it with a default - implementation which is based on environment variables -

Reply:
`configuration-resolver.config-files` allow us to inject multiple files into the system and each file could point to a specific config-provider - so we could fetch credentials from multiple system and multiple instances of same system.

Reply:
> What do you think about making this file be TOML similar to https://github.com/airlift/airlift/pull/1189? Then we could support multiple providers in a single file.
We could try - But I saw @martint had a few comments regarding its conversion to `Map<String, String>` - How are we planning to represent them ?